src/idlcxx/src/streamers.c

```diff
@@ -388,20 +476,33 @@ unroll_array(
   uint32_t array_depth,
   bool is_key)
 {
-  const char* afmt = array_depth ? "  for (%1$sauto & a_%2$u:a_%3$u)\n"
-                                 : "  for (%1$sauto & a_%2$u:%4$s)\n";
-
-  if (putf(&streams->write, afmt, "const ", array_depth+1, array_depth, accessor)
-   || putf(&streams->read, afmt, "", array_depth+1, array_depth, accessor)
-   || putf(&streams->move, afmt, "const ", array_depth+1, array_depth, accessor)
-   || putf(&streams->max, afmt, "const ", array_depth+1, array_depth, accessor))
-    return IDL_RETCODE_NO_MEMORY;
-  if (is_key &&
-      (putf(&streams->key_write, afmt, "const ", array_depth+1, array_depth, accessor)
-    || putf(&streams->key_read, afmt, "", array_depth+1, array_depth, accessor)
-    || putf(&streams->key_move, afmt, "const ", array_depth+1, array_depth, accessor)
-    || putf(&streams->key_max, afmt, "const ", array_depth+1, array_depth, accessor)))
-    return IDL_RETCODE_NO_MEMORY;
+  if (array_depth) {
+    const char* afmt = "  for (%1$sauto & a_%2$u:a_%3$u)\n";
+    if (putf(&streams->write, afmt, "const ", array_depth+1, array_depth)
+      || putf(&streams->read, afmt, "", array_depth+1, array_depth)
+      || putf(&streams->move, afmt, "const ", array_depth+1, array_depth)
+      || putf(&streams->max, afmt, "const ", array_depth+1, array_depth))
+      return IDL_RETCODE_NO_MEMORY;
+    if (is_key &&
+      (putf(&streams->key_write, afmt, "const ", array_depth+1, array_depth)
+        || putf(&streams->key_read, afmt, "", array_depth+1, array_depth)
+        || putf(&streams->key_move, afmt, "const ", array_depth+1, array_depth)
+        || putf(&streams->key_max, afmt, "const ", array_depth+1, array_depth)))
```

Comment:
Indentation differs throughout the whole file.

Reply:
Will go through the file and unify this.

Reply:
Have fixed this.